packages/eslint-plugin/src/util/misc.ts

```diff
@@ -242,20 +242,19 @@ type NodeWithKey =
 function getStaticMemberAccessValue(
   node: NodeWithKey,
   { sourceCode }: RuleContext<string, unknown[]>,
-): string | undefined {
+): string | null {
   const key =
     node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
   if (!node.computed) {
     return key.type === AST_NODE_TYPES.Literal
       ? `${key.value}`
       : (key as TSESTree.Identifier | TSESTree.PrivateIdentifier).name;
   }
-  const value = getStaticValue(key, sourceCode.getScope(node))?.value as
-    | string
-    | number
-    | null
-    | undefined;
-  return value == null ? undefined : `${value}`;
+  const result = getStaticValue(key, sourceCode.getScope(node));
+  /* If a value is returned, coerce it to a string, because that's what JavaScript does with any data type in a
```

Comment:
Well, JS doesn't do this with symbols, though (which is kind of the whole point of symbols). Symbols are keys unto themselves. So, I wouldn't think it would make sense to stringify symbols ever (their string representation is only a debugging tool). Shouldn't we just return the symbol itself?

Reply:
Yes, I agree that is more correct. The only downside is that Symbol-based methods will not be able to be allow-listed using things like `explicit-module-boundary-types` => `allowedNames` or `class-methods-use-this` => `exceptMethods`.